tools/donate-cpu-server.py

```diff
@@ -52,14 +56,17 @@ def handle_uncaught_exception(exc_type, exc_value, exc_traceback):
 
 
 def strDateTime():
+    # type: () -> str
     return datetime.datetime.now().strftime('%Y-%m-%d %H:%M')
 
 
 def dateTimeFromStr(datestr):
```

Comment:
How about `def dateTimeFromStr(datestr: str):`

Reply:
AFAIK this would require at least Python 3.5. I am not sure if everyone who develops on this script has such a recent version. Personally I like these typeinfo hints.

Reply:
> AFAIK this would require at least Python 3.5. I am not sure if everyone who develops on this script has such a recent version. Personally I like these typeinfo hints.

That is not quite right. The function annotations itself are available since Python 3.0, see https://www.python.org/dev/peps/pep-3107/.
Type hints (see https://www.python.org/dev/peps/pep-0484/) are available since Python 3.5.
So it looks like it is acceptable to add function annotations like you suggested (and I also like better).

Reply:
I have added Python 3.0 function annotations instead of the comments with type hints now.
For the client script, which should be Python 2 compatible, we could use the comments with type hints. What do you think?

Reply:
yes in the client script the comments are fine.